gossipd/routing.c

```diff
@@ -636,19 +726,20 @@ const struct short_channel_id *handle_channel_announcement(
 		     type_to_string(pending, struct short_channel_id,
 				    &pending->short_channel_id));
 
-	/* FIXME: Handle duplicates as per BOLT #7 */
+	/* So you're new in town, ey? Let's find you a room in the Inn. */
+	if (chan == NULL)
+		chan = routing_channel_new(chan, &pending->short_channel_id);
```

Comment:
The new `chan` will have a `NULL` tal parent? Is this deliberate?  If so, should it not be better `NULL` expicitly?

Reply:
Good catch...

Reply:
Maybe it is better to make `rstate` the parent?  Since, routing channels are intended, to be in the map that `rstate` holds?

Reply:
Will fixup and make `rstate` the parent.